### ISSUE TITLE
/etc/hosts gets conflicting entries for guest VM brought up with vagrant

### ISSUE THREAD

**Issue by jbw976** [Parallels/vagrant-parallels#263]
I have the following versions installed on my Mac OS X:
vagrant-parallels (1.6.2)
Parallels Desktop 11 for Mac Business Edition, Version 11.2.0 (32581)

With the following Vagrantfile, using a box built from https://coreos.com/releases/#1032.1.0 with no modifications and is published for anyone to use freely:
```
Vagrant.configure(2) do |config|
  config.vm.box = "jbw976/coreos-alpha"

  config.vm.define "coreos-alpha" do |node|
    node.vm.network :private_network, type: "dhcp"
  end
end
```

After a vagrant up, the hosts /etc/hosts file has 2 entries for the hostonly adapter and 2 entries for the shared adapter, each with a different IP address:
```
> cat /etc/hosts
127.0.0.1	localhost
255.255.255.255	broadcasthost
::1             localhost 
10.37.129.77   	parallels-etc-hosts-repro-coreos-alpha-1463099075889-57114.host-only parallels-etc-hosts-repro-coreos-alpha-1463099075889-57114 #prl_hostonly host-only
10.211.55.248  	parallels-etc-hosts-repro-coreos-alpha-1463099075889-57114.shared parallels-etc-hosts-repro-coreos-alpha-1463099075889-57114 #prl_hostonly shared
10.37.129.80   	parallels-etc-hosts-repro-coreos-alpha-1463099075889-57114.host-only parallels-etc-hosts-repro-coreos-alpha-1463099075889-57114 #prl_hostonly host-only
10.211.55.249  	parallels-etc-hosts-repro-coreos-alpha-1463099075889-57114.shared parallels-etc-hosts-repro-coreos-alpha-1463099075889-57114 #prl_hostonly shared
```

The guest VM only shows the 10.211.55.249 address and the 10.37.129.80 address:
```
core@parallels-etc-hosts-repro_coreos-alpha_1463099075889_57114 /etc/systemd/network $ ifconfig
eth0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 10.211.55.249  netmask 255.255.255.0  broadcast 10.211.55.255
        inet6 fe80::21c:42ff:fe6a:96bf  prefixlen 64  scopeid 0x20<link>
        inet6 fdb2:2c26:f4e4:1:21c:42ff:fe6a:96bf  prefixlen 64  scopeid 0x0<global>
        inet6 fdb2:2c26:f4e4:0:21c:42ff:fe6a:96bf  prefixlen 64  scopeid 0x0<global>
        ether 00:1c:42:6a:96:bf  txqueuelen 1000  (Ethernet)
        RX packets 3850  bytes 341415 (333.4 KiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 2217  bytes 235019 (229.5 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

eth1: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 10.37.129.80  netmask 255.255.255.0  broadcast 10.37.129.255
        inet6 fdb2:2c26:f4e4:0:21c:42ff:fe0f:f52f  prefixlen 64  scopeid 0x0<global>
        inet6 fdb2:2c26:f4e4:1:21c:42ff:fe0f:f52f  prefixlen 64  scopeid 0x0<global>
        inet6 fe80::21c:42ff:fe0f:f52f  prefixlen 64  scopeid 0x20<link>
        ether 00:1c:42:0f:f5:2f  txqueuelen 1000  (Ethernet)
        RX packets 52  bytes 8380 (8.1 KiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 119  bytes 14464 (14.1 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        inet6 ::1  prefixlen 128  scopeid 0x10<host>
        loop  txqueuelen 1  (Local Loopback)
        RX packets 12  bytes 996 (996.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 12  bytes 996 (996.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
```

Using the DNS name to reach the guest VM does not work due to the conflicting entries in /etc/hosts.

**Comment by jbw976** [comment]
Also including a tcpdump capture while the vagrant up is running, let me know if different filters/flags are needed for a better capture.
[output-tcpdump.txt](https://github.com/Parallels/vagrant-parallels/files/262473/output-tcpdump.txt)

**Comment by jbw976** [comment]
@dskr99, @bassamtabbara mentioned this may be of interest to you.  Let me know of any other information I can provide.  Thanks!

**Comment by legal90** [comment]
Hi @jbw976,
Thank you for reporting this issue! We have fixed it, so the fix will be available in the next update of Parallels Desktop. It should be released approximately in June-July 2016. I'll let you know here when it happens.

**Comment by jbw976** [comment]
That's great news @legal90, thanks for the update.  Do you have a link to the commit/fix, or is the fix in closed source? I'm curious to know what the problem is, for my own learning sake.

Thanks!

**Comment by legal90** [comment]
@jbw976 The issue you've faced with is pretty close to this one: https://github.com/mitchellh/packer/issues/1746

After the first boot, your CoreOS VM has changed its DHCP ID (DUID) by some reason. It causes internal Parallels DHCP server to lease yet another IP for the same network interface of the VM. In your case, it happens with both of VM's network interfaces: Shared (default, pre-configured in the box) and Host-Only (configured via `private_network ` in Vagrantfile).

So that, there are more than one IP addresses leased by internal DHCP server for the same MAC (all records could be found in `/Library/Preferences/Parallels/parallels_dhcp_leases`). 

The problem was fixed on the Parallels Desktop side, which is closed source. 
It just makes the feature of auto-filling `/etc/hosts` work as expected: in this file, there should be only the one, actual IP address for each VM's interface.

**Comment by jbw976** [comment]
@legal90 @dskr99 is it possible to get access with our business license to a pre-release installer with the fix?  or do you know of a potential workaround?

the issue seems to be a bit more broad than just conflicting entries in the /etc/hosts file, as DNS requests within the guest VM also resolve to the 1st stale/old address.  Note below that eth1 has 10.37.129.96 but DNS response from parallels host-only network says 10.37.129.97:
```
core@registry ~
> ifconfig
...
eth1: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 10.37.129.96  netmask 255.255.255.0  broadcast 10.37.129.255
...

> nslookup registry
Server:		10.37.129.1
Address:	10.37.129.1#53

Name:	registry
Address: 10.37.129.97
```

**Comment by dskr99** [comment]
the only possible workaround I can suggest for the issue, is to call "sudo killall -HUP prl_naptd" - this will rewrite the file /etc/hosts and remove the duplicated entry.. but this of course applies only when you know that there is a duplicated IP

**Comment by jbw976** [comment]
@dskr99 that workaround seems to work just fine for our scenario/environment.  Thanks for the suggestion! looking forward to the permanent fix to be released.

**Comment by jbw976** [comment]
I just tried this scenario again after updating to Parallels Desktop 11 for Mac Business Edition Version 11.2.1 (32626), and it appears fixed.  Can you confirm that the fix has been released in that version upstream?  Thanks!

note the suggested workaround has been working well for us this whole time.

**Comment by legal90** [comment]
@jbw976 Yes, confirmed - this issue is fixed in Parallels Desktop 11.2.1.

**Comment by jbw976** [comment]
@legal90, great news, thanks for getting the fix out!
